src/views/Home.vue

```diff
@@ -314,50 +344,31 @@ export default {
     },
     openUploader () {
       this.files = []
+      this.uploadPhotos = []
       this.uploadCounter = 0
       this.show = true
-      document.getElementById('uploadPanel').innerHTML = ''
     },
-    generateUploadImgElement (fileObj) {
-      var file = fileObj.file
-      var id = fileObj.id
-      var reader = new FileReader()
-      reader.onload = (function (file, id) {
-        // debugger
-        return function (e) {
-          document.getElementById('uploadPanel').innerHTML +=
-            ['<div id="', id, '" style="display:inline-block;position:relative"><img id="img-', id, '" src="', e.target.result, '" class="upload-preview upload-preview-pending"/></div>'].join('')
-        }
-      })(file, id)
+    generateUploadPhotoElement (fileObj) {
+      const self = this
+      const reader = new FileReader()
+      const file = fileObj.file
+      const id = fileObj.id
 
-      reader.readAsDataURL(file, id)
-    },
-    updateUploadImgStatus (fileObj) {
-      var file = fileObj.file
-      var id = fileObj.id
-      var reader = new FileReader()
-      reader.onload = (function (file, id) {
-        // debugger
-        return function (e) {
-          document.getElementById(id).innerHTML +=
-            ['<div id="spin-', id, '"class="spin"><i class="fa fa-spinner fa-spin"/></div>'].join('')
+      reader.addEventListener('load', () => {
+        const file = event.target
+        const photo = {
+          id: id,
+          src: file.result,
+          status: 'pending'
         }
-      })(file, id)
+        self.uploadPhotos.push(photo)
+      })
 
-      reader.readAsDataURL(file, id)
-    },
-    updateUploadImgCompleted (fileObj) {
-      var id = fileObj.id
-      var imgId = ['img-', id].join('')
-      document.getElementById(imgId).classList.remove('upload-preview-pending')
-      var spinId = ['spin-', id].join('')
-      var spin = document.getElementById(spinId)
-      spin.parentNode.removeChild(spin)
+      reader.readAsDataURL(file)
     },
-    updateUploadImgFailed (fileObj) {
-      var id = fileObj.id
-      document.getElementById(id).innerHTML +=
-            ['<div id="failed-', id, '"class="spin"><i class="fas fa-ban"/></div>'].join('')
+    updateUploadPhotoStatus (photoId, status) {
+      const index = this.uploadPhotos.findIndex(photo => photo.id === photoId)
```

Comment:
why not ```this.uploadPhotos.find(...).status = status```

Reply:
Good idea. But I think I should check whether `this.uploadPhotos.find(...)` exist to avoid js error. What do you think if I change the code to

```
const index = this.uploadPhotos.findIndex(photo => photo.id === photoId)
if (index < 0) {
  return
}
this.uploadPhotos[index].status = status
```